koa-basic-auth/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for koa-basic-auth 2.x
+// Project: https://github.com/koajs/basic-auth
+// Definitions by: Tobias Wolff <https://github.com/Tobias4872/>
+// Definitions: https://github.com/Tobias4872/DefinitelyTyped
+
+import * as Koa from "koa";
+
+declare function auth(opts?: {
+
+    name: string;
+
+    pass: string;
+
+}): { (ctx: Koa.Context, next?: () => any): any };
```

Comment:
Sorry, were do you see the definition of "Koa.Middleware"? I have scanned both the Koa package as well as the Koa declarations, but no luck.